docs/descriptor.rst

```diff
@@ -155,9 +155,31 @@ correctly. On Windows, it is recommended that you use forward slashes.
 
 .. Note:: When using the git descriptor, you need to have the git executable in
           the ``PATH`` in order for the API to be able to do a latest check or
-          app download. The git exeuctable is, however, not needed during descriptor
+          app download. The git executable is, however, not needed during descriptor
           resolve and normal operation.
 
+Toolkit does not support git credentials. If you are using authenticated git repositories,
+you need to configure these externally. We strongly recommend SSH-style urls for your
+repositories, e.g. on the form ``git@github.com:shotgunsoftware/tk-config-default.git``. When using
+these, the operating system will handle all authentication.
+
+If you have to use http or https based git urls with credentials, you may get an error from Toolkit
+looking something like this::
+
+    TankDescriptorError: Could not get tags for https://github.com/shotgunsoftware/tk-config-xyz.git:
+    Error executing git operation 'git ls-remote --tags "https://github.com/shotgunsoftware/tk-config-xyz.git"':
+    bash: /dev/tty: No such device or address
```

Comment:
i have now refactored the code to behave like it did previously, meaning that this doc goes away.